### PR TITLE
Helmchart integration test

### LINKED ISSUE #37: Fix the Helm chart and add a chart deployment test
The Helm chart can have an integration test deployment to a kind cluster. This should test that the latest tag works along with metrics collection from a prometheus instance.
Additionally, the chart needs a number of fixes including the selector in the Deployment/Service and working cert-manager certificates creation for metrics and the webhook.

### PR DESCRIPTION
**Pull request by trevorbox**
Closes #37

### CODE REVIEW

File: .github/workflows/pr.yaml [hunk 1]
```diff
@@ -8,7 +8,8 @@ on:
 jobs:
   shared-operator-workflow:
     name: shared-operator-workflow
-    uses: redhat-cop/github-workflows-operators/.github/workflows/pr-operator.yml@v1.0.1
+    uses: trevorbox/github-workflows-operators/.github/workflows/pr-operator.yml@helmchart-test
```
Comment (trevorbox) [inline .github/workflows/pr.yaml:11]: Waiting on https://github.com/redhat-cop/github-workflows-operators/pull/9

**Comment by raffaelespazzoli** [comment]
please see if we can use this approach https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/api.md#tlsconfig

File: config/prometheus/monitor.yaml [hunk 2]
```diff
@@ -14,7 +14,11 @@ spec:
       port: https
       scheme: https
       tlsConfig:
-        caFile: /etc/prometheus/configmaps/serving-certs-ca-bundle/service-ca.crt
+        ca: 
+          secret:
+            key: 'tls.crt'
+            name: vault-config-operator-certs
+            optional: false
```
Comment (trevorbox) [inline config/prometheus/monitor.yaml:21]: @raffaelespazzoli for some reason I didn't need to give the prometheus serviceaccount permission (update the Role in the rbac config section) to read Secrets in the namespace. Tested successfully using kind and in OCP after a bundle deploy.

Comment (trevorbox) [reply config/prometheus/monitor.yaml:21]: Sounds like the operator has access to the Secrets, prometheus itself doesn't need rbac to Secrets.

**Comment by trevorbox** [comment]
@raffaelespazzoli this is ready for review

**Comment by raffaelespazzoli** [comment]
should fix #44

**Pull request merged**